x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/document_fields/fields/field_types/completion_type.tsx

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import { EuiSpacer } from '@elastic/eui';
+import { NormalizedField, Field as FieldType } from '../../../../types';
+import { getFieldConfig } from '../../../../lib';
+import { UseField, Field, CheckBoxField, FormDataProvider } from '../../../../shared_imports';
+import { AnalyzerParameter } from '../../field_parameters';
+import { EditFieldSection, EditFieldFormRow, AdvancedSettingsWrapper } from '../edit_field';
+
+const getDefaultValueToggle = (param: string, field: FieldType) => {
+  switch (param) {
+    case 'max_input_length': {
+      return field[param] !== undefined && field[param] !== getFieldConfig(param).defaultValue;
+    }
+    case 'analyzers': {
+      return field.search_analyzer !== undefined && field.search_analyzer !== field.analyzer;
+    }
+    default:
+      return false;
+  }
+};
+
+interface Props {
+  field: NormalizedField;
+}
+
+export const CompletionType = ({ field }: Props) => {
+  return (
+    <AdvancedSettingsWrapper>
+      <EditFieldSection>
+        {/* analyzers */}
+        <EditFieldSection
+          title={i18n.translate('xpack.idxMgmt.mappingsEditor.completion.analyzersSectionTitle', {
+            defaultMessage: 'Analyzers',
+          })}
+        >
+          <AnalyzerParameter
+            path="analyzer"
+            label={i18n.translate(
```

Comment:
I think here we need the same logic as in Text type where the label changes from "Index + search analyzer" to "Index analyzer" according to the checkbox.

But... there is a regression in Text type and the change of label was removed in some refactoring. So we need to do it in both places. 😊

Reply:
Fixed via https://github.com/elastic/kibana/pull/51429